app/main.py

```diff
@@ -1,25 +1,140 @@
-from abc import ABC
+from __future__ import annotations
+from abc import ABC, abstractmethod
 
 
 class IntegerRange:
-    pass
+    def __init__(
+            self,
+            min_amount: int,
+            max_amount: int
+    ) -> None:
+        self.min_amount = min_amount
+        self.max_amount = max_amount
+
+    def __set_name__(
+            self,
+            owner: SlideLimitationValidator,
+            name: str
+    ) -> None:
+        self.private_name = "_" + name
+        self.public_name = name
+
+    def __get__(
+            self,
+            instance: SlideLimitationValidator,
+            owner: Slide
+    ) -> int:
+        return getattr(instance, self.private_name)
+
+    def __set__(
+            self,
+            instance: SlideLimitationValidator,
+            value: int
+    ) -> None:
+        if instance.validate(value, self.min_amount, self.max_amount):
+            setattr(instance, self.private_name, value)
+        else:
+            setattr(instance, self.private_name, False)
 
 
 class Visitor:
-   pass
+    def __init__(
+        self,
+        name: str,
+        age: int,
+        weight: int,
+        height: int
+    ) -> None:
+        self.name = name
+        self.age = age
+        self.weight = weight
+        self.height = height
 
 
 class SlideLimitationValidator(ABC):
-    pass
+    def __init__(
+            self,
+            age: int,
+            weight: int,
+            height: int
+    ) -> None:
+        self.age = age
+        self.weight = weight
+        self.height = height
+
+    @abstractmethod
+    def validate(
+            self,
+            value: int,
+            min_value: int,
+            max_value: int
+    ) -> bool:
+        return isinstance(value, int) and min_value <= value <= max_value
```

Comment:
Do not declare and implement the `validate` method here and in the other classes, rely on the logic of the `__set__` method.